homeassistant/components/backup/manager.py

```diff
@@ -244,12 +244,22 @@ async def _async_upload_backup(
     ) -> None:
         """Upload a backup to selected agents."""
         LOGGER.warning("Uploading backup %s to agents %s", backup.backup_id, agent_ids)
+
+        async def send_backup() -> AsyncGenerator[bytes]:
+            f = await self.hass.async_add_executor_job(path.open, "rb")
+            with f:
```

Comment:
Closing the file, exiting the context, will make a syscall I think, which we don't want to do in async context.

Reply:
Also, is the context enter a no-op since the file is already open?

Reply:
Fixed